### PR TITLE
Fixed broken test_random_choice

### PR DESCRIPTION
**Pull request by vfdev-5**
`test_random_choice` is broken on main:
- https://github.com/pytorch/vision/actions/runs/4254956199/jobs/7401929827#step:10:3954

Description:
- Fixed `test_random_choice` and `test_asserts`. 
- Improved error message in RandomChoice

### CODE REVIEW

**Review by NicolasHug — COMMENTED** [review]
Thanks for the PR @vfdev-5 , the changes to RandomChoice LGTM, however I don't think we should mark the other tests as `xfail`. 

Those tests are passing on most jobs, they're only failing on 3.10. But the problem isn't the tests in themselves, the tests are merely *surfacing* a (packaging?) issue that is tracked in https://github.com/pytorch/vision/issues/7299, and which hopefully the releng team will be able to address soon.

I'd prefer keeping those tests in to make sure our warnings work as expected, and ignore the 3.10 signals for now

**Comment by NicolasHug** [comment]
In fact the 3.9 and 3.8 jobs are now failing because we use *strict* xfail: pytest expects those tests to fail but they actually pass, so it raises an error/

File: test/test_transforms_v2.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import pathlib
 import random
 import re
+import sys
```
Comment (NicolasHug) [inline test/test_transforms_v2.py:5]: Not needed :) ?

**Review by NicolasHug — APPROVED** [review]
Thanks @vfdev-5 !

File: test/test_transforms_v2_consistency.py [hunk 2]
```diff
@@ -822,7 +822,7 @@ def test_random_choice(self, probabilities):
                 v2_transforms.Resize(256),
                 legacy_transforms.CenterCrop(224),
             ],
-            probabilities=probabilities,
+            p=probabilities,
```
Comment (pmeier) [inline test/test_transforms_v2_consistency.py:825]: Nit: should we also rename the parameter to `p`?

```suggestion
            p=p,
```

File: test/test_transforms_v2.py (hunk 1 above)
Comment (vfdev-5) [reply test/test_transforms_v2.py:5]: Thanks !

**Review by pmeier — APPROVED** [review]
Thanks!

File: test/test_transforms_v2_consistency.py (hunk 2 above)
Comment (vfdev-5) [reply test/test_transforms_v2_consistency.py:825]: I think it is ok

**Pull request merged**